firedrake/matrix_free/operators.py

```diff
@@ -266,6 +266,7 @@ def multTranspose(self, mat, Y, X):
                 self._x += self._xbc
         else:
             # No DirichletBC/EquationBC
+            # There is only a single element in the list (for the domain equation).
```

Comment:
I think it is safer to assert that explicitly, e.g. by:

```
aT, = self._assemble_actionT
aT()
```

or else just loop over them (in case something changes).

Reply:
This looks nice! Now fixed.